### PR TITLE
specify teal minimal version

### LINKED ISSUE #11: [Question]: Unable to to launch sample apps beyond basic-teal
### What is your question?

I am discovering the teal packages and wanted to try locally the apps using the teal gallery.

I install the packages and dependencies but when I am trying to launch apps with teal.gallery::launch_app() I have the following error.
The error is present in all app except basic-teal.

> teal.gallery::list_apps()
[1] "basic-teal"      "early-dev"       "efficacy"        "exploratory"     "longitudinal"    "patient-profile" "python"          "RNA-seq"         "safety"         
> teal.gallery::launch_app("safety")
Finding R package dependencies ... Done!
[INFO] 2023-02-22 14:39:19.0994 pid:18429 token:[] teal.modules.general Initializing tm_front_page
Error in module(label = label, server = srv_front_page, ui = ui_front_page,  : 
  
`filters = NULL` indicates that the module doesn't need any data while it has the `data` in formals.
Please specify `filters` with the names of needed datasets or exclude `data` from the arguments.

For reference below if my sessionInfo()
```
R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS 13.0.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] teal.goshawk_0.1.14.9009          goshawk_0.1.14.9008               stringr_1.5.0                     DescTools_0.99.48                 sparkline_2.0                    
 [6] nestcolor_0.1.0.9005              teal.osprey_0.1.14                osprey_0.1.15                     teal.modules.general_0.2.15.9016  shinyTree_0.2.7                  
[11] ggmosaic_0.3.3                    ggplot2_3.4.1                     teal.modules.clinical_0.8.14.9019 tern_0.7.10                       rtables_0.5.3.4                  
[16] formatters_0.3.4.19               dplyr_1.1.0                       scda.2022_0.1.3.9016              scda_0.1.5.9007                   teal_0.12.0                      
[21] teal.transform_0.2.0              magrittr_2.0.3                    teal.data_0.1.2                   shiny_1.7.4                      

loaded via a namespace (and not attached):
  [1] colorspace_2.1-0          ellipsis_0.3.2            class_7.3-21              mmrm_0.2.2.9010           estimability_1.4.1        gld_2.6.6                
  [7] rstudioapi_0.14           proxy_0.4-27              remotes_2.4.2             DT_0.27                   ggrepel_0.9.3             fansi_1.0.4              
 [13] mvtnorm_1.1-3             splines_4.2.0             cachem_1.0.6              rootSolve_1.8.2.3         knitr_1.42                jsonlite_1.8.4           
 [19] mcr_1.3.2                 broom_1.0.3               geepack_1.3.9             compiler_4.2.0            httr_1.4.4                emmeans_1.8.4-1          
 [25] backports_1.4.1           assertthat_0.2.1          Matrix_1.5-3              fastmap_1.1.0             lazyeval_0.2.2            cli_3.6.0                
 [31] later_1.3.0               htmltools_0.5.4           tools_4.2.0               coda_0.19-4               gtable_0.3.1              glue_1.6.2               
 [37] lmom_2.9                  Rcpp_1.0.10               cellranger_1.1.0          jquerylib_0.1.4           vctrs_0.5.2               nlme_3.1-162             
 [43] teal.logger_0.1.1         crosstalk_1.2.0           xfun_0.37                 rbibutils_2.2.13          mime_0.12                 lifecycle_1.0.3          
 [49] renv_0.16.0               staged.dependencies_0.2.8 MASS_7.3-58.2             scales_1.2.1              promises_1.2.0.1          parallel_4.2.0           
 [55] tern.gee_0.1.0            expm_0.999-7              yaml_2.3.7                curl_5.0.0                Exact_3.2                 memoise_2.0.1            
 [61] teal.slice_0.2.0          robslopes_1.1.2           teal.widgets_0.2.0        sass_0.4.5                stringi_1.7.12            e1071_1.7-13             
 [67] checkmate_2.1.0           teal.reporter_0.1.1       boot_1.3-28.1             teal.gallery_0.1.0.9003   Rdpack_2.4                rlang_1.0.6              
 [73] pkgconfig_2.0.3           evaluate_0.20             lattice_0.20-45           fontawesome_0.5.0         purrr_1.0.1               htmlwidgets_1.6.1        
 [79] cowplot_1.1.1             tidyselect_1.2.0          logger_0.2.2              R6_2.5.1                  generics_0.1.3            pillar_1.8.1             
 [85] withr_2.5.0               survival_3.5-3            tibble_3.1.8              crayon_1.5.2              shinyWidgets_0.7.6        utf8_1.2.3               
 [91] plotly_4.10.1             rmarkdown_2.20            readxl_1.4.2              grid_4.2.0                data.table_1.14.8         digest_0.6.31            
 [97] tern.mmrm_0.2.1           xtable_1.8-4              numDeriv_2016.8-1.1       tidyr_1.3.0               httpuv_1.6.9              munsell_0.5.0            
[103] viridisLite_0.4.1         bslib_0.4.2               shinyjs_2.1.0       
```

Thaks for the help!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by pawelru**
an attempt to fix #11

I think the last "global" change was shinyvalidate. I derieved min versions from linked PRs to https://github.com/insightsengineering/NEST-roadmap/issues/57. Please correct me if I missed something.
It's a shortcut solution. More correct would be to implement min version of teal in tmg and others.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -47,14 +47,14 @@ Suggests:
     scda.2022 (>= 0.1.3),
     sparkline,
     stringr,
-    teal (>= 0.12.0),
+    teal (>= 0.12.0.9012),
```
Comment (gogonzo) [inline DESCRIPTION:50]: current @main version is `.9022`. Are we suppose to keep the current version or last "breaking" change? 

We are going to merge https://github.com/insightsengineering/teal/pull/806 today so I suggest to even upgrade to `.9023`

**Comment by gogonzo** [comment]
I've checked locally - using latest main everywhere fixes the apps. I suggest to keep this as a starting point and upgrade when some new features are added.

**Comment by gogonzo** [comment]
@pawelru I propose following changes:
- teal.gallery having only `teal.modules.* (>= x.x.x.xxxx)` in the DESCRIPTION
- `teal.modules.*` packages should have `teal (>= x.x.x.xxxx)`, `teal.transform (>= x.x.x.xxxx)`, ``teal.code (>= x.x.x.xxxx)`` in the DESCRIPTION and respective widgets
- `teal` should have `teal.slice (>= x.x.x.xxxx)`, `teal.transform (>= x.x.x.xxxx)`, etc.

I don't think we should change version of teal.slice (for example) everywhere - only in the packages which really depends on it (teal)

**Comment by pawelru** [comment]
This is not that easy :) It needs to also have (at least) `teal`. Basically the key is that every package with a direct call needs to be enumerated in the DESCRIPTION file. 

The difficulty here is that there are `Depends` (as opposed to `Imports`) down in the dependency tree. This makes some of the packages seems to be obsolete in DESC here but I think we still need them. Great example is `teal`. `teal.modules.*` has `teal` it its Depends. Let's assume that it changes its api significantly, `teal.modules.*` is backwards compatible so no need to update DESC there but `teal.gallery` is not so we have to have dedicated entry for `teal` in order to be able to specify min version. I hope you got what I mean.

**Comment by gogonzo** [comment]
> This is not that easy :) It needs to also have (at least) `teal`. Basically the key is that every package with a direct call needs to be enumerated in the DESCRIPTION file.

I agree. I can also say that it's fair and safe to move `teal` from suggests to imports 

> Great example is teal. teal.modules.* has teal it its Depends. Let's assume that it changes its api significantly, teal.modules.* is backwards compatible so no need to update

Not really, all `teal.modules.*` have teal in Depends but we control minimal version there - which means that if teal changes API then `teal.modules.*` will have DESCRIPTION changed.

> The difficulty here is that there are `Depends` (as opposed to `Imports`) down in the dependency tree. This makes some of the packages seems to be obsolete in DESC here but I think we still need them. 

I don't think we need them. If something is not explicitly called in teal.gallery then it could be solved within other package. Let's list possible breaking solutions:

### 1. lower version of teal

Let's imagine `"basic-teal"`. We have `teal` lower version, so the app runs as it suppose to for the lower version and with any version of installed teal.

### 2. lower version of teal for teal.modules.general

Let's assume one want to run `"exploratory"` with the old version of `teal.slice`. Then `teal.modules.general` will throw error on load because of insufficient `teal` for `tmg`, or insufficient `teal.slice` for `teal`.

### 3. mismatch between teal.modules.general and teal.modules.clinical

Let's assume someone has en old version of `tmg` and new of `tmc`. If one was able to install new `tmc`, this means that newest `teal.slice` (through `teal`) version has been installed also. The app will fail on `runApp` as `tmg` will return some error related to the change in the `teal.slice` API. This scenario can be fixed with setting minimal required versions of `teal.modules.*` in `teal.gallery`.

### 4. higher version of teal.slice and lower of teal

Let's assume somebody installed a new `teal.slice` version and kept old `teal`. Let's also assume that it's a breaking change - all teal apps fail. We can't prevent this situation neither by putting `teal.slice (>= )` in teal nor in teal.modules.* - because each higher version of `teal.slice` matche the `>=` condition. This is scenario which we can't avoid but it can be fixed by reinstalling - as reinstall always takes latest available version.

**Comment by pawelru** [comment]
My previous message was about (theoretically) breaking change in e.g. `teal` that is fully compatible with `teal.modules.*` but it's not with `teal.gallery`.

Let's say we have `teal::foo <- function(arg1, ...) {...}`. In `teal.modules.*` we have `teal::foo(1, ...)` and in `teal.gallery` - `teal::foo(arg1 = 1, ...)`. Now we do a breaking change in teal about renaming argument `arg1` into `arg_1`. `teal.modules.*` is totally fine as it doesn't name the arguments. Therefore there is no need to vbump dependency. But obviously we need to change `teal.gallery` and vbump deps. 

This would be addressed when we list all packages with a direct calls. I think we both agree on that so please proceed.

**Review by donyunardi — APPROVED** [review]
LGTM.

**Pull request merged**